--- a/iptables/xtables-arp.c
+++ b/iptables/xtables-arp.c
@@ -437,10 +437,6 @@
 		exit(1);
 	}
 
-#if defined(ALL_INCLUSIVE) || defined(NO_SHARED_LIBS)
-	init_extensionsa();
-#endif
-
 	if (nft_init(h, NFPROTO_ARP, xtables_arp) < 0)
 		xtables_error(OTHER_PROBLEM,
 			      "Could not initialize nftables layer.");
--- a/iptables/xtables-eb.c
+++ b/iptables/xtables-eb.c
@@ -684,10 +684,6 @@
 		exit(1);
 	}
 
-#if defined(ALL_INCLUSIVE) || defined(NO_SHARED_LIBS)
-	init_extensionsb();
-#endif
-
 	if (nft_init(h, NFPROTO_BRIDGE, xtables_bridge) < 0)
 		xtables_error(OTHER_PROBLEM,
 			      "Could not initialize nftables layer.");
